test/spec/modules/smaatoBidAdapter_spec.js

```diff
@@ -368,222 +102,537 @@ describe('smaatoBidAdapterTest', () => {
+
+        expect(req_without_usp.regs.ext.us_privacy).to.not.exist;
+      });
+
+      it('sends first party data', () => {
+        sinon.sandbox.create().stub(config, 'getConfig').callsFake(key => {
+          const config = {
+            ortb2: {
+              site: {
+                keywords: 'power tools,drills'
+              },
+              user: {
+                keywords: 'a,b',
+                gender: 'M',
+                yob: 1984
               }
-            ]
-          },
-          tagid: 'adspaceId'
-        }
-      ])
-    });
+            }
+          };
+          return utils.deepAccess(config, key);
+        });
+        const bidRequest = utils.deepClone(singleBannerBidRequest);
+
+        const req_fpd = JSON.parse(spec.buildRequests([bidRequest], defaultBidderRequest).data);
+
+        expect(req_fpd.user.gender).to.equal('M');
+        expect(req_fpd.user.yob).to.equal(1984);
+        expect(req_fpd.user.keywords).to.eql('a,b');
+        expect(req_fpd.user.ext.consent).to.equal(CONSENT_STRING);
+        expect(req_fpd.site.keywords).to.eql('power tools,drills');
+        expect(req_fpd.site.publisher.id).to.equal('publisherId');
+      });
 
-    it('sends correct site', () => {
-      expect(this.req.site.id).to.exist.and.to.be.a('string');
-      expect(this.req.site.domain).to.exist.and.to.be.a('string');
-      expect(this.req.site.page).to.exist.and.to.be.a('string');
-      expect(this.req.site.ref).to.equal('http://example.com/page.html');
-      expect(this.req.site.publisher.id).to.equal('publisherId');
-    })
+      it('has no user ids', () => {
+        const req = JSON.parse(spec.buildRequests([singleBannerBidRequest], defaultBidderRequest).data);
 
-    it('sends gdpr applies if exists', () => {
-      expect(this.req.regs.ext.gdpr).to.equal(1);
-      expect(this.req.user.ext.consent).to.equal('HFIDUYFIUYIUYWIPOI87392DSU');
+        expect(req.user.ext.eids).to.not.exist;
+      });
     });
 
-    it('sends no gdpr applies if no gdpr exists', () => {
-      let req_without_gdpr = JSON.parse(spec.buildRequests([singleBannerBidRequest], minimalBidderRequest).data);
-      expect(req_without_gdpr.regs.ext.gdpr).to.not.exist;
-      expect(req_without_gdpr.user.ext.consent).to.not.exist;
-    });
+    describe('multiple impressions', () => {
+      it('build individual request for each impression', () => {
+        const bidRequest1 = utils.deepClone(singleBannerBidRequest);
+        const bidRequest1BidId = '1111'
+        utils.deepSetValue(bidRequest1, 'bidId', bidRequest1BidId);
+        const bidRequest2 = utils.deepClone(singleBannerBidRequest);
+        const bidRequest2BidId = '2222'
+        utils.deepSetValue(bidRequest2, 'bidId', bidRequest2BidId);
 
-    it('sends usp if exists', () => {
-      expect(this.req.regs.ext.us_privacy).to.equal('uspConsentString');
-    });
+        const reqs = JSON.parse(spec.buildRequests([bidRequest1, bidRequest2], defaultBidderRequest).data)
```

Comment:
if I understood it all correctly this is not multi-imp, it is two separate bid requests.

we should have one bidRequest with two imp inside

Reply:
yeah you are right, completely overlooked that